### PR TITLE
Secrets manager

### PR DESCRIPTION
**Pull request by twratl**

### CODE REVIEW

File: src/britive/britive.py [hunk 1]
```diff
@@ -180,7 +184,14 @@ def patch_upload(self, url, file_content_as_str, content_type, filename):
...
             return response.content.decode('utf-8')
-
+            
+    def post_upload(self, url, params=None, files=None):
+        """Internal use only."""
+        response = self.session.post(url, params=params, files=files, headers={'Content-Type': None})
+        try:
+            return response.json()
+        except native_json.decoder.JSONDecodeError:  # if we cannot decode json then the response isn't json
+            return response.content.decode('utf-8')
```
Comment (twratl) [inline src/britive/britive.py:195]: space this out

File: src/britive/notification_mediums.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+class NotificationMediums:
+    def __init__(self, britive):
+        self.britive = britive
+        self.base_url = f'{self.britive.base_url}/v1/notification-service/notificationmediums'
+    def list(self):
+        """
+        Provide a list of all notification mediums
```
Comment (twratl) [inline src/britive/notification_mediums.py]: List all notification mediums.

Comment (twratl) [reply src/britive/notification_mediums.py]: with newline then :return

File: src/britive/notification_mediums.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        params = {'name': name, 'description': description, 'type': type, 'connectionParameters': connectionParameters}
+        return(self.britive.post(self.base_url, json=params))
+    def get(self, notification_medium_id):
+        """
+        Provide details of the given notification medium, from a notification medium id.
+        :param notification_medium_id: The ID  of the notification medium.
+        :return: Details of the specified notification medium.
+        """
+        return self.britive.get(f'{self.base_url}/{notification_medium_id}')
+    def update(self, notification_medium_id : str, **kwargs):
```
Comment (twratl) [inline src/britive/notification_mediums.py]: connectionParameter: dict = {}

File: src/britive/notification_mediums.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        self.britive = britive
+        self.base_url = f'{self.britive.base_url}/v1/notification-service/notificationmediums'
+    def list(self):
+        """
+        Provide a list of all notification mediums 
+        :return: List of all notification mediums
+        """
+        return self.britive.get(self.base_url)
+
+    def create(self, type : str, name : str, description : str = "Default notification medium description", connectionParameters : dict = {}) -> dict:
```
Comment (twratl) [inline src/britive/notification_mediums.py]: check line lengths (<=120 characters)

File: src/britive/notification_mediums.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            name : the name of the notification medium
+            description : the description of the notification medium
+            connectionParameters : the connection parameters of the notification medium
+                valid connection parameters: 
+                    URL : for slack, the URL
+                    token : for slack, Auth Token for the Application BOT
+                    Webhook URL : for teams, the URL of the teams webhook
+        :return: Details of the newly created notification medium.
+        """
+        params = {'name': name, 'description': description, 'type': type, 'connectionParameters': connectionParameters}
```
Comment (twratl) [inline src/britive/notification_mediums.py]: make multi line

File: src/britive/secrets_manager.py [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+        self.base_url = f'{self.britive.base_url}/v1/secretmanager/pwdpolicies'
+    def get(self, password_policy_id : str):
+        """
+        Provide details of the given password policy, from a password policy id.
+        :param password_policy_id: The ID  of the password policy.
+        :return: Details of the specified password policy.
+        
+        """
+        return self.britive.get(f'{self.base_url}/{password_policy_id}')
+        return self.britive.get(self.base_url, params=params)
```
Comment (twratl) [inline src/britive/secrets_manager.py]: clean up?

File: src/britive/notification_mediums.py (hunk 2 above)
Comment (Turtdle) [reply src/britive/notification_mediums.py]: Fixed in 637defb2698c5507cf1d0bc494707e7badd46d13

File: src/britive/notification_mediums.py (hunk 4 above)
Comment (Turtdle) [reply src/britive/notification_mediums.py]: fixed in 637defb2698c5507cf1d0bc494707e7badd46d13

File: src/britive/notification_mediums.py (hunk 5 above)
Comment (Turtdle) [reply src/britive/notification_mediums.py]: fixed in 637defb2698c5507cf1d0bc494707e7badd46d13

File: src/britive/britive.py (hunk 1 above)
Comment (Turtdle) [reply src/britive/britive.py:195]: finished in 80810d7247c7a5e1e78a5765ab466861bb53a39c

File: src/britive/secrets_manager.py (hunk 6 above)
Comment (Turtdle) [reply src/britive/secrets_manager.py]: finished in 80810d7247c7a5e1e78a5765ab466861bb53a39c

**Pull request merged**